akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1011,10 +1012,10 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
    * This object can be cached and reused within the same [[GraphStageLogic]].
    */
   final def getAsyncCallback[T](handler: T ⇒ Unit): AsyncCallback[T] = {
-    val result = new ConcurrentAsyncCallback[T](handler)
-    asyncCallbacksInProgress.add(result)
-    if (_interpreter != null) result.onStart()
-    result
+    val callback = new ConcurrentAsyncCallback[T](handler)
+    if (_interpreter != null) callback.onStart()
+    else callbacksWaitingForInterpreter = callback :: callbacksWaitingForInterpreter
```

Comment:
ok, so first they are added to that `callbacksWaitingForInterpreter`, which is drained in `beforePreStart`

after that `onStart()` is used, which will use the currentState (Pending -> Initializing -> Initialized)